mgwr/utils.py

```diff
@@ -1,34 +1,44 @@
 import numpy as np
 from libpysal.common import requires
+from matplotlib.colors import LinearSegmentedColormap
+import pandas as pd
 
 
 @requires('matplotlib')
-def shift_colormap(cmap, start=0, midpoint=0.5, stop=1.0, name='shiftedcmap'):
-    '''
-    Function to offset the "center" of a colormap. Useful for
-    data with a negative min and positive max and you want the
-    middle of the colormap's dynamic range to be at zero
-
-    Parameters
-    ----------
-    cmap : The matplotlib colormap to be altered
-    start : Offset from lowest point in the colormap's range.
-      Defaults to 0.0 (no lower ofset). Should be between
-      0.0 and `midpoint`.
-    midpoint : The new center of the colormap. Defaults to
-      0.5 (no shift). Should be between 0.0 and 1.0. In
-      general, this should be  1 - vmax/(vmax + abs(vmin))
-      For example if your data range from -15.0 to +5.0 and
-      you want the center of the colormap at 0.0, `midpoint`
-      should be set to  1 - 5/(5 + 15)) or 0.75
-    stop : Offset from highets point in the colormap's range.
-      Defaults to 1.0 (no upper ofset). Should be between
-      `midpoint` and 1.0.
-    
-    Returns
-    -------
-    new_cmap : A new colormap that has been shifted. 
-    '''
+def shift_colormap(cmap: LinearSegmentedColormap,
```

Comment:
Are you sure that only `LinearSegmentedColormap` is allowed? not other types of Colormaps?

Reply:
Hi @martinfleis,

I've reviewed the existing codebase and found that LinearSegmentedColormap is the only colormap utilized in MGWR. If I've overlooked any additional colormaps currently in use, your assistance in identifying them would be greatly appreciated.